src/utils/constraints.js

```diff
@@ -43,6 +43,38 @@ export const createQuestion = {
   }
 };
 
+export const editProfile = {
+  email(_, attributes) {
+    if (
+      !attributes.currentPassword &&
+      !attributes.newPassword &&
+      !attributes.newPasswordConfirmation
+    ) {
+      return {
+        presence: { message: 'edit.form.presence' }
+      };
+    }
+  },
+  password(_, attributes) {
+    if (attributes.currentPassword) {
```

Comment:
Is this correct? Or it should be `!attributes.currentPassword`? If you validate presence only if the value is present, you are not actually validating anything 😆

Reply:
Lol that's right, but this is not the case. Here, currentPassword refers to the user's current password atm, while password and passwordConfirmation refer to the new password the user is trying to set. Thus, password field should only be required if the user has already written something at currentPassword field.

I understand password should be better called newPassword to avoid confusion, but this is actually the key the backend expects so 🤷‍♂️

Reply:
That makes sense 👍